src/components/ProductDetail/ProductDetailOrderForm.js

```diff
@@ -0,0 +1,138 @@
+import { useEffect, useState } from "react";
+import SizeOptionsContainer from "./SizeOptionsContainer";
+import { camelToPascalCase } from "../../helperFunctions";
+import { Link } from "react-router-dom";
+
+const ProductDetailOrderForm = ({product, addToShoppingBag, shoppingBag, updateQuantity}) => {
+  const [isSingleSize, setIsSingleSize] = useState(false);
+  const [inputFields, setInputFields] = useState({
+    "color":"",
+    "size": "",
+    "quantity":0
+  });
+  const [colorOptions, setColorOptions] = useState([]);
+  const [isFormHealthy, setIsFormHealthy] = useState(false);
+  const [totalQuantity, setTotalQuantity] = useState(0);
+  
+  useEffect(() => {
+    setColorOptions(product?.colorOptions)
+  }, [product]);
+  
+  useEffect(() => {
+    checkFormHealth()
+  }, [inputFields]);
+
+  useEffect(() => {
+    console.log(isFormHealthy)
+  }, [isFormHealthy]);
+
+  useEffect(() => {
+    const total = shoppingBag.reduce((total, item) => total + item.quantity, 0);
+    setTotalQuantity(total);
```

Comment:
there is already a function that does this that rachel made in app if you wanted to dry up the code, but it is a very basic function so doesn't reallllly matter that it's here again

Reply:
Good catch. I missed that but I see that it's setting a state in App (which we wouldn't want it here). So it would be tricky to use it.

Reply:
ah